mico-core/src/main/java/io/github/ust/mico/core/dto/GitHubCrawlingInfoDTO.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.github.ust.mico.core.dto;
+
+import javax.validation.constraints.NotEmpty;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+import io.github.ust.mico.core.configuration.extension.CustomOpenApiExtentionsPlugin;
+import io.github.ust.mico.core.service.GitHubCrawler;
+import io.swagger.annotations.ApiModelProperty;
+import io.swagger.annotations.Extension;
+import io.swagger.annotations.ExtensionProperty;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.experimental.Accessors;
+
+/**
+ * DTO for the information needed by the {@link GitHubCrawler}
+ * for crawling a service from a GitHub repository.
+ */
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Accessors(chain = true)
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class GitHubCrawlingInfoDTO {
+
+    /**
+     * The url to the GitHub repository to crawl from.
+     */
+    @ApiModelProperty(required = true, extensions = {
+            @Extension(name = CustomOpenApiExtentionsPlugin.X_MICO_CUSTOM_EXTENSION, properties = {
+                    @ExtensionProperty(name = "title", value = "URL"),
+                    @ExtensionProperty(name = "x-order", value = "10"),
+                    @ExtensionProperty(name = "description", value = "The url to the GitHub repository to crawl from.") }) })
+    @NotEmpty
+    private String url;
+
+    /**
+     * The GitHub release tag. Defaults to 'latest'.
+     */
+    @ApiModelProperty(extensions = {
+            @Extension(name = CustomOpenApiExtentionsPlugin.X_MICO_CUSTOM_EXTENSION, properties = {
+                    @ExtensionProperty(name = "title", value = "Version"),
+                    @ExtensionProperty(name = "x-order", value = "20"),
+                    @ExtensionProperty(name = "description", value = "The GitHub release tag. Defaults to 'latest'.") }) })
+    @NotEmpty
```

Comment:
If you think the `@NotEmpty` could be misunderstood that it is some kind of required, add a commend like:
`Must not be {@code null} but can be omitted (default value will be used)`